backend/app/services/chat/trigger/unified.py

```
@@ -278,6 +282,24 @@ async def _process_contexts(
 
+    # Build KB meta prompt for dynamic_context injection in chat_shell.
+    # Keeping system prompts static improves prompt caching.
+    request.kb_meta_prompt = ""
+    if request.task_id:
+        try:
+            from app.services.chat.preprocessing.kb_meta import (
+                build_kb_meta_prompt_for_task,
+            )
+
+            request.kb_meta_prompt = build_kb_meta_prompt_for_task(db, request.task_id)
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)